pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -475,6 +483,27 @@ func endpointScrapeConfig(id string, ep ScrapeEndpoint, relabelCfgs []*relabel.C
 		},
 	}
 
+	// Use the pod name as the primary identifier in the instance label. Unless the pod
+	// is controlled by a DaemonSet, in which case the node name will be used.
+	// This provides a better user experience on dashboards which template on the instance label
```

Comment:
So if there are two DaemonSets being monitored:
- `example-ds1` with labels `app=ds1`, `app.kubernetes.io/managed-by=example-controller` emitting on port name `metrics`
- `example-ds2` with labels `app=ds2`, `app.kubernetes.io/managed-by=example-controller` emitting on port name `metrics`

Something like `instance=k8s-node-1:metrics` would capture both targets, with the `job` target label being different, assuming the two scrape configs were generated by two PodMonitorings; one scraping using selector `app=ds1`, and the other `app=ds2`.

However, if there is one PodMonitoring that captures both DaemonSets, i.e. by using selector `app.kubernetes.io/managed-by=example-controller`, then would the targets be conflicting and merged together somehow? Or am I overlooking something? Maybe `honor_labels` disambiguation takes over here?

Also, if this relabling is ok, then this seems like behavior we should surface in our api documentation. What do you think about adding a note to https://github.com/GoogleCloudPlatform/prometheus-engine/blob/b9b52f970feb4788eaf4c36c7f04f02d19dc5007/pkg/operator/apis/monitoring/v1/types.go#L857-L858 and https://github.com/GoogleCloudPlatform/prometheus-engine/blob/b9b52f970feb4788eaf4c36c7f04f02d19dc5007/pkg/operator/apis/monitoring/v1/types.go#L886-L887 something like:

```
If the label selector selects pods controlled by a DaemonSet, the generated `instance` label will contain the Kubernetes node name of the embedded pod rather than the pod name itself.
```

Reply:
You are right, good analysis. This could indeed be a conflict. By default we'd still add the `pod` label however, which would distinguish them again.

In this particular case I'd say that the benefit of `instance` having a more meaningful value outweighs the collision risk, which is fairly minimal I think and contained within a single PodMonitorig, i.e. not users affecting each other.

Reply:
Docs added on `TargetLabels` as I suspect that's where users will look to understand how they are created.